torch/_functorch/_aot_autograd/subclass_utils.py

```diff
@@ -165,16 +258,28 @@ def wrap_tensor_subclasses(
 ) -> Tuple[Any, ...]:
     wrapped_args = []
     num_args_tallied = 0
+    num_args = sum(
+        1 if isinstance(meta, int) else meta.arg_count for meta in subclass_metas
+    )
+    count_extra = len(unwrapped_args) > num_args
     for subclass_meta in subclass_metas:
         if isinstance(subclass_meta, int):
             wrapped_args.append(unwrapped_args[subclass_meta])
             num_args_tallied += 1
         else:
             assert isinstance(subclass_meta, SubclassCreationMeta)
             wrapped_args.append(
-                subclass_meta.creation_fn(unwrapped_args, is_runtime=is_runtime)
+                subclass_meta.creation_fn(
+                    unwrapped_args, is_runtime=is_runtime, is_subclass=False
+                )
             )
+            assert subclass_meta.outer_size is not None
             num_args_tallied += subclass_meta.arg_count
+            num_args_tallied = (
```

Comment:
hmm. Why can't we just ensure that `subclass_meta.arg_count` is set properly to include the extra size args, so we don't need to do anything special here?

Reply:
Done! This simplified the code quite a bit.